### PR TITLE
configure Loki for local development

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Extend our docker compose setup to include Loki (along with promtail), scraping `logs/*.log` to ingest development logs as well as all logs from running containers. While we're in here, teach Prometheus to scrape metrics from Docker containers too.

Right now, Loki isn't a default service, so to enable be sure to define something like this before running the server:
```
ENABLED_DOCKER_SERVICES=postgres inbucket openldap grafana prometheus loki promtail
```

You'll also want to enable logging to disk (as currently implemented, Promtail scrapes this file and sends to Loki), either in `config.json` or via:
```
MM_LOGSETTINGS_ENABLEFILE=true
MM_LOGSETTINGS_FILELEVEL=debug
MM_LOGSETTINGS_FILEJSON=true
MM_LOGSETTINGS_FILELOCATION=logs
```

And do a `make stop-docker start-docker` to reload all the related configuration. Then browse to Grafana at http://localhost:3000/ and explore!

Those of you not using Docker can run Loki, Grafana, and promtail on the host using the same basic configuration files.

Those of you using OrbStack will be frustrated by https://github.com/orbstack/orbstack/issues/1148, but hopefully this gets resolved soon.

#### Ticket Link
None

##### Mattermost
CleanShot 2024-05-13 at 11 59 46@2x

##### Inbucket
CleanShot 2024-05-13 at 12 00 17@2x

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by lieut-data** [comment]
@agarciamontoro & @agnivade, this works for me on MacOS, but I'd love your 👀 from the Linux world (assuming you let the `Makefile` manage your Docker containers!)

I'm hoping we can get this into the hands of developers to boost confidence using Loki, and help develop experience once the work to expose Cloud production logs completes.

**Comment by agnivade** [comment]
Nice! I'll give this a spin. 

cc @sadohert as well because we were discussing this some time back.

**Review by agnivade — APPROVED** [review]
Just tested it. Worked flawlessly! :100: 

Worth mentioning that live mode is a treat. Just watch your logs flowing as you use the application.

File: server/build/docker/promtail/promtail-local-config.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          selector: '{job="mattermost"}'
+          stages:
+            - json:
+                expressions:
+                  timestamp: timestamp
+                  level: level
+      - labels:
+          level:
+      - timestamp:
+          format: '2006-01-02 15:04:05.999 -07:00'
```
[the remark this answers is not in the document]
Comment (lieut-data) [reply server/build/docker/promtail/promtail-local-config.yaml:24]: Ah, fascinating. Only squinting here, but reading up on [time.Parse](https://pkg.go.dev/time#Parse) it looks like some layouts can include both the `Z` and a timezone offset, e.g.:
```go
	// Some valid layouts are invalid time values, due to format specifiers
	// such as _ for space padding and Z for zone information.
	// For example the RFC3339 layout 2006-01-02T15:04:05Z07:00
	// contains both Z and a time zone offset in order to handle both valid options:
	// 2006-01-02T15:04:05Z
	// 2006-01-02T15:04:05+07:00
	t, _ = time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
	fmt.Println(t)
	t, _ = time.Parse(time.RFC3339, "2006-01-02T15:04:05+07:00")
	fmt.Println(t)
```

I picked `-07:00` as I /thought/ that was the way to capture the timezone correctly from the log, but maybe it just happened to work in my case? Does something like `2006-01-02 15:04:05.999 Z07:00` work in your case? (Also, do you know what led to that particular log output, as I'd love to reproduce it.)

[the remark this answers is not in the document]
Comment (streamer45) [reply server/build/docker/promtail/promtail-local-config.yaml:24]: I am running our official 9.6.0 release. I think the only difference is that it's running on an AWS instance which has its local time set to UTC.

But of course, now I can't reproduce the original issue any longer by reverting my changes so maybe it was a temporary glitch that had nothing to do with the timestamp format parsing :thinking:

**Pull request merged**